lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -119,6 +119,7 @@ def create
     t = Tempfile.new('openldap_database')
     t << "dn: olcDatabase=#{resource[:backend]},cn=config\n"
     t << "changetype: add\n"
+    t << "objectClass: olcDatabaseConfig\n"
```

Comment:
Higher in the file (in the [self.instances](https://github.com/OpenConceptConsulting/puppet-openldap/blob/0cd415ca3415492765f66da24aa1c355f931c4a3/lib/puppet/provider/openldap_database/olc.rb#L19) method), the database objects are required to have the `olcDatabaseConfig` objectclass. 

AFAIK, that object class only has one MUST attribute, `olcDatabase`, which is being specified.